docker/contrib/python/image.bzl

```diff
@@ -0,0 +1,171 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A rule for creating a Python Docker image.
+
+The signature of this rule is compatible with py_binary.
+"""
+
+load(
+    "//docker:build.bzl",
+    _build_attrs = "attrs",
+    _build_implementation = "implementation",
+    _build_outputs = "outputs",
+)
+load("//docker:pull.bzl", "docker_pull")
+
+def _dep_layer_impl(ctx):
+  """Appends a layer for a single dependencies runfiles."""
+
+  return _build_implementation(
+    ctx,
+    # We put the files from dependency layers into a binary-agnostic
+    # path to increase the likelihood of layer sharing across images,
+    # then we symlink them into the appropriate place in the app layer.
+    # This references the binary package because the file paths are
+    # relative to it, and normalized by the tarball package.
+    directory="/app/" + ctx.label.package,
+    files=list(ctx.attr.dep.default_runfiles.files),
+    symlinks={
+      # Handle empty files by linking to /dev/null
+      "/app/" + empty: "/dev/null"
+      for empty in ctx.attr.dep.default_runfiles.empty_filenames
+    }
+  )
+
+_dep_layer = rule(
+    attrs = _build_attrs + {
+        # The base image on which to overlay the dependency layers.
+        "base": attr.label(default = Label("@py_image_base//image")),
```

Comment:
That's a good question.

For the latter, I think this is just up to the toolchain that `rules_go` pulls in, and I punt on importing the Go rules in this repo (they must be imported before `contrib/go/image.bzl` or it will fail).

For the former, I haven't done anything specifically to address this, but I think the story should be one of:
1. Change the `base` to an appropriate image (requires exposing `base`, which has a TODO now)
1. Make the rules smart about `PY2`, `PY3`, `PY2AND3` in its choice of the distroless base (and perhaps override-able by the above) .

I would love to hear your thoughts and/or clever ideas here.